bjs2 41281 ban users

ZhubanyshZh opened the pull request.

ZhubanyshZh committed:
feat(41281): ban users

add method banUsers which ban users by users ids
and add banned field to User model

ZhubanyshZh committed:
feat(41281): ban users

add method redis config and listener for ban users and fixing some moment during testing

MiraGlaceon commented on src/main/java/school/faang/user_service/config/redis/RedisConfig.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    @Bean
+    ChannelTopic recommendationTopic() {
```
Модификаторы доступа забыл. Работает, но не очень красиво, давай соблюдать код стайл

MiraGlaceon commented on src/main/java/school/faang/user_service/config/redis/RedisConfig.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    @Bean
+    MessageListenerAdapter usersForBanReceivedEventListenerAdapter(UsersForBanReceivedEventListener usersForBanReceivedEventListener) {
```
У тебя не так много топиков, но все равно глянь, команда нашла подход как избавиться от излишнего объявления бинов топиков https://github.com/CorporationX/analytics_service/pull/363

MiraGlaceon commented on src/main/java/school/faang/user_service/service/user/BannerService.java:
```diff
@@ -0,0 +1,38 @@
+        List<User> bannedUsers = new ArrayList<>();
+        userRepository.findAllByIds(userIds)
+                .ifPresentOrElse(
+                        users -> users.stream()
```
Конструкция запутывает. Лучше кусок кода, где пользователям выставляется признак banned вынести в отдельный метод. И где используется список bannedUsers?

MiraGlaceon commented on src/main/java/school/faang/user_service/service/user/BannerService.java:
```diff
@@ -0,0 +1,38 @@
+                                .toList(),
+                        () -> {
+                            log.info("users by ids: {}, not found!", userIds);
+                            return;
```
Здесь без return можно. И так конец метода

MiraGlaceon commented on src/main/resources/db/changelog/changeset/user_V011__update_users_add_banned.sql:
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE users
+    ADD COLUMN IF NOT EXISTS banned boolean
```
Не замечание, больше совет: именуй булевые переменные с припиской "is", например, "is_banned". То же и с переменными в коде

ZhubanyshZh committed:
fix(41281): RedisConfig,BannerService

Refactored RedisConfig to handle topics dynamically and simplify listener configuration. Improved BannerService by separating user banning logic into a dedicated method for better readability and maintainability

ZhubanyshZh committed:
test(41281): BannerUserTest

Introduced test cases to verify user banning logic, ensuring both user retrieval and ban status updates are correctly handled.

ZhubanyshZh committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-41281-ban-users

# Conflicts:
#	src/main/java/school/faang/user_service/config/redis/RedisConfig.java
#	src/main/resources/application.yaml

ZhubanyshZh committed:
fix imports
